app/src/main/java/com/example/footprnt/Map/Util/MapUtil.java

```diff
@@ -179,4 +197,149 @@ public static JSONObject getContinents(Activity activity){
         return mContinents;
     }
 
+    /**
+     * On search result, locates place on map
+     *
+     * @param searchText
+     * @param map
+     * @param context
+     */
+    public static void geoLocate(EditText searchText, GoogleMap map, Context context){
+        String searchString = searchText.getText().toString();
+
+        Geocoder geocoder = new Geocoder(context);
+        List<Address> list = new ArrayList<>();
+        try{
+            list = geocoder.getFromLocationName(searchString, 1);
+        } catch (IOException e){
+
+        }
+
+        if (list.size() > 0){
+            Address address = list.get(0);
+            Location l = new Location(LocationManager.GPS_PROVIDER);
+            l.setLatitude(address.getLatitude());
+            l.setLongitude(address.getLongitude());
+            centreMapOnLocation(map, l);
+        }
+    }
+
+    /**
+     * Centres camera on given location
+     *
+     * @param map
+     * @param location
+     */
+    public static void centreMapOnLocation(GoogleMap map, Location location) {
```

Comment:
This is a little overkill in terms of encapsulation. A good rule of thumb would be to think about everything that can go wrong with this method, if you can handle that with a single IF statement (in this case just checking whether `location` is null), it's probably not complicated enough to be its own method.